### PR TITLE
change default profiles dir in dbt examples

### PR DESCRIPTION
**Pull request by sryza**
This makes the examples cleaner, and also lets people run the dbt projects by just going into the project directories and running `dbt build` etc.

DbtCli will automatically look for the profiles.yml inside the root of the dbt projects.

### CODE REVIEW

File: examples/assets_dbt_python/dbt_project/profiles.yml
Comment (rexledesma) [inline examples/assets_dbt_python/dbt_project/profiles.yml:1]: `profiles.yml` was move to the top level, but we should also just remove the `config/` directory in the dbt project. The user id shouldn't be checked in anyways.

**Review by rexledesma — CHANGES_REQUESTED** [review]
This will not work out of the box since `load_assets_from_dbt_project` needs to be compatible with `DbtCli`.

We will need to branch on the underlying op logic, depending on what resource was passed in.

File: examples/assets_dbt_python/dbt_project/profiles.yml
Comment (sryza) [reply examples/assets_dbt_python/dbt_project/profiles.yml:1]: oops I missed that - thanks for catching

**Review by rexledesma — COMMENTED** [review]
This change should just be merged with the previous PR since `DbtCli` doesn't accept `profiles_dir` in the first place

**Comment by sryza** [comment]
closing and smushing into https://github.com/dagster-io/dagster/pull/15036

**Comment by sryza** [comment]
because they can't be merged independently

**Pull request closed without merging**